### PR TITLE
[DDS-1324] Added configuration for temporary files on mounted volume

### PR DESCRIPTION
**Pull request by nicksantamaria**
https://digital-vic.atlassian.net/browse/DDS-1324

## Problem

Currently Drupal's temporary directory is `/tmp` which means it is not shared between pods. The impact of this is that large webform submission data exports have missing records as the batch job has divergent caches as the web requests are load balanced.

## Solution

Move the temporary storage to the mounted filesystem. This will allow pods to share the temporary files.

## Technical notes

This PR adds support for a new environment variable:

- `BAY_SHARED_TEMP_FILES` - setting to `"false"` will disable shared temp files, and revert to using `/tmp`.

### CODE REVIEW

File: images/bay-php/settings.php [hunk 1]
```diff
@@ -242,9 +242,16 @@
...
+if (empty($settings['file_private_path'])) {
+  $settings['file_private_path'] = 'sites/default/files/private';
+}
+
+// Use mounted fileshare for temp directory unless explicity disabled with
+//  BAY_SHARED_TEMP_FILES=false
+if (getenv("BAY_SHARED_TEMP_FILES") !== "false") {
+  $config['system.file']['path']['temporary'] = sprintf("%s/tmp", $settings['file_private_path']);
+} else {
+  $config['system.file']['path']['temporary'] = getenv("TMPDIR") ?: "/tmp";
```
Comment (anthony-malkoun) [inline images/bay-php/settings.php:254]: Sensible defaults 👏

Comment (anthony-malkoun) [reply images/bay-php/settings.php:254]: Although the deleted line refers to `TMP` rather than `TMPDIR`?

**Review by anthony-malkoun — APPROVED** [review]
Thanks Nick, keen to get this in 😍

File: images/bay-php/settings.php (hunk 1 above)
Comment (nicksantamaria) [reply images/bay-php/settings.php:254]: Your comment prompted me to dig into this - https://en.wikipedia.org/wiki/TMPDIR

Looks like TMPDIR is the most widely used in linux/unix systems.

**Pull request merged**